### DISCUSSION TITLE
Why language server - auto completion and linting not working

### DISCUSSION THREAD

**Discussion by Gr-Nobel** [cpeditor/cpeditor#1117] [Q&A] [answered]
I installed this version and added to the path.
![image](https://user-images.githubusercontent.com/95166364/203483319-7a3ec5f8-ab85-474a-a554-3b45b3aaa03b.png)

Then changed the prefereces 
![image](https://user-images.githubusercontent.com/95166364/203483377-f0a3ee76-28b2-499e-adb7-29f5b5dd919c.png)

But the linting and auto completion no working. NO popup while typing code.
![image](https://user-images.githubusercontent.com/95166364/203483550-daa30e25-9d07-4df5-b867-3158a824c13b.png)

even wrong red lines on ok code lines.

**Comment by ouuan** [comment, marked as answer]
CP Editor does not support auto-completion yet.

The red lines are caused by `clangd` unable to find the `iostream` header. Using `clangd` on Windows is troublesome, and I'm not familiar with it. You can try using the LLVM bundled with CP Editor (download a CP Editor release asset with "LLVM" in its filename).

**Reply by Gr-Nobel** [reply]
I downloaded the LLMV from their website is it not ok?

**Reply by ouuan** [reply]
> I downloaded the LLMV from their website is it not ok?

I don't know, but you can try my solution. Maybe there are other solutions using the LLVM you downloaded, but I just do not know.

**Comment by ouuan** [comment]
CP Editor bundles [winlibs](https://github.com/brechtsanders/winlibs_mingw/releases), you may also download GCC and LLVM there.
